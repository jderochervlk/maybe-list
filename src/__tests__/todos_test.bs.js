// Generated by ReScript, PLEASE EDIT WITH CARE
'use strict';

var Jest = require("@glennsl/rescript-jest/src/jest.bs.js");
var Curry = require("rescript/lib/js/curry.js");
var Todos = require("../Todos.bs.js");
var React = require("react");
var Caml_array = require("rescript/lib/js/caml_array.js");
var ReactTestingLibrary = require("../external/ReactTestingLibrary.bs.js");

var todos = [
  {
    text: "one",
    completed: false,
    createdAt: 1.0,
    id: 1
  },
  {
    text: "two",
    completed: false,
    createdAt: 2.0,
    id: 2
  }
];

Jest.test("toggle todo", (function (param) {
        var results = Todos.toggleTodo([
              {
                text: "one",
                completed: false,
                createdAt: 1.0,
                id: 1
              },
              {
                text: "two",
                completed: false,
                createdAt: 2.0,
                id: 2
              }
            ], {
              text: "one",
              completed: false,
              createdAt: 1.0,
              id: 1
            });
        return Jest.Expect.toEqual(Jest.Expect.expect(results), [
                    {
                      text: "two",
                      completed: false,
                      createdAt: 2.0,
                      id: 2
                    },
                    {
                      text: "one",
                      completed: true,
                      createdAt: 1.0,
                      id: 1
                    }
                  ]);
      }));

Jest.test("renders", (function (param) {
        var result = ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, React.createElement(Todos.make, {
                  todos: todos
                }));
        return Jest.Expect.toMatchSnapshot(Jest.Expect.expect(result.container));
      }));

Jest.test("updates on click", (function (param) {
        var result = ReactTestingLibrary.render(undefined, undefined, undefined, undefined, undefined, React.createElement(Todos.make, {
                  todos: [Caml_array.get(todos, 0)]
                }));
        ReactTestingLibrary.act(function (param) {
              Curry._2(ReactTestingLibrary.FireEvent.click, undefined, ReactTestingLibrary.getByRole({
                        NAME: "Str",
                        VAL: "checkbox"
                      }, undefined, result));
            });
        return Jest.Expect.toMatchSnapshot(Jest.Expect.expect(result.container));
      }));

exports.todos = todos;
/*  Not a pure module */
