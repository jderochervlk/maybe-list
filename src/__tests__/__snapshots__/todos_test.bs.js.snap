// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <ul>
    <li
      class=""
    >
      <input
        type="checkbox"
      />
      one
    </li>
    <li
      class=""
    >
      <input
        type="checkbox"
      />
      two
    </li>
  </ul>
</div>
`;

exports[`updates on click 1`] = `
<div>
  <ul>
    <li
      class="completed"
    >
      <input
        checked=""
        type="checkbox"
      />
      one
    </li>
  </ul>
</div>
`;
